packages/@aws-cdk/aws-lambda/lib/function-base.ts

```diff
@@ -283,6 +290,14 @@ export abstract class FunctionBase extends Resource implements IFunction, ec2.IC
     const action = permission.action ?? 'lambda:InvokeFunction';
     const scope = permission.scope ?? this;
 
+    if (['lambda:InvokeFunction', 'lambda:*'].includes(action) && !qualified) {
```

Comment:
Is the scope of this narrow enough? I am concerned we might end up giving unnecessary warnings to customers. But, I can't think of a way to narrow this down further (without crazy amounts of effort).

Reply:
I think since `lambda:*` gives permission to `lambda:InvokeFunction` we have to give the warning since it's possible that these customers are broken by the new change. I think we should consider adding `lambda:Invoke*` as well to the list (even though there's no other `lambda:InvokeXxx` permission, which is why I didn't include it in the first place).